version: "3.7"

services:

  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - gis_5ban_network
    volumes:
      - /home/ubuntu/nginx.conf:/etc/nginx/nginx.conf
      - gis_5ban_static:/data/staticfiles/
      - gis_5ban_media:/data/media/

  django:
    image: gis_5ban_django:4
    networks:
      - gis_5ban_network
    volumes:
      - gis_5ban_static:/home/GJAI_web_tea/staticfiles/
      - gis_5ban_media:/home/GJAI_web_tea/media/
    secrets:
      - DJANGO_SECRET_KEY

  mariadb:
    image: mariadb:10.5
    networks:
      - gis_5ban_network
    volumes:
      - gis_mariadb:/var/lib/mysql
    environment:
      MARIADB_ROOT_PASSWORD: 1q2w3e4r!
      MARIADB_DATABASE: django
      MARIADB_USER: django
      MARIADB_PASSWORD: 1q2w3e4r!


networks:
  gis_5ban_network:

volumes:
  gis_5ban_static:
  gis_5ban_media:
  gis_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true